LoopKit/GlucoseRangeSchedule+SafeBounds.swift

```diff
@@ -0,0 +1,34 @@
+//
+//  GlucoseRangeSchedule+SafeBounds.swift
+//  LoopKit
+//
+//  Created by Noah Brauner on 12/5/22.
+//  Copyright © 2022 LoopKit Authors. All rights reserved.
+//
+
+import Foundation
+import HealthKit
+
+extension GlucoseRangeSchedule {
+    public func safeSchedule(with suspendThreshold: GlucoseThreshold?) -> GlucoseRangeSchedule? {
+        var glucoseScheduleMinimunValue: Double!
+        if let threshold = suspendThreshold {
+            glucoseScheduleMinimunValue = Guardrail.minCorrectionRangeValue(suspendThreshold: threshold).doubleValue(for: threshold.unit)
+        }
+        else {
+            glucoseScheduleMinimunValue = Guardrail.correctionRange.absoluteBounds.lowerBound.doubleValue(for: .milligramsPerDeciliter)
+        }
+        return rangeScheduleWithMin(min: glucoseScheduleMinimunValue)
+    }
+
+    private func rangeScheduleWithMin(min: Double) -> GlucoseRangeSchedule? {
+        let filteredItems = rangeSchedule.valueSchedule.items.map { scheduleValue in
+            let newScheduleValue = DoubleRange(minValue: max(min, scheduleValue.value.minValue), maxValue: max(min, scheduleValue.value.maxValue))
```

Comment:
The units of the value passed in here, `min`, are mg/dL.  The scheduleValue.value items are going to be in units specified by `unit`.

Reply:
Thanks for catching that, I just fixed it. Also, the safeSchedule function will now only be called in onboarding, as its not necessary to check if a user is just modifying their settings.